server/src/MessageHandler.ts

```diff
@@ -107,3 +191,100 @@ export async function handleDocumentFormatting(
     TextEdit.replace(fullDocumentRange(document), formatted),
   ])
 }
+
+/**
+ *
+ * This handler provides the initial list of the completion items.
+ */
+export function handleCompletionRequest(
+  params: CompletionParams,
+  documents: TextDocuments<TextDocument>,
+): CompletionList | undefined {
+  const context = params.context
+  if (context == null) {
```

Comment:
Please use triple equality by default for everything.
It's a common pitfall of javascript, note that `undefined == null`.